### ISSUE TITLE
rcdefaults, rc_file_defaults, rc_file should not update backend if it has already been selected

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#9135]
### Bug report

**Bug summary**
```
matplotlib.use("qt5agg")
matplotlib.rcdefaults()
print(matplotlib.get_backend()) # i.e. rcParams["backend"]
```
**Code for reproduction**

See above.

**Actual outcome**

`plt.get_backend()` returns "Agg".

**Expected outcome**

`plt.get_backend()` returns the correct backend (which did not change).

**Matplotlib version**

  * Operating System: Linux
  * Matplotlib Version: master but likely around since ever
  * Python Version: 3.6
  * Jupyter Version (if applicable):
  * Other Libraries:

**Comment by tacaswell** [comment]
:+1: anything we blacklist from styles should probably be immune to `rcdefaults` as well.

**Comment by anntzer** [comment]
Closed by #9150.
